aws/session/shared_config.go

```diff
@@ -308,6 +338,30 @@ func (cfg *sharedConfig) setFromIniFiles(profiles map[string]struct{}, profile s
 		cfg.SourceProfile = srcCfg
 	}
 
+	// If the profile contains an SSO session parameter, the session MUST exist
+	// as a section in the config file. Load the SSO session using the name
+	// provided. If the session section is not found or incomplete an error
+	// will be returned.
+	if cfg.hasSSOTokenProviderConfiguration() {
+		skippedFiles = 0
+		for _, f := range files {
+			section, ok := f.IniData.GetSection(fmt.Sprintf(ssoSectionPrefix + strings.TrimSpace(cfg.SSOSessionName)))
+			if ok {
+				var ssoSession SSOSession
+				ssoSession.setFromIniSection(section)
+				ssoSession.Name = cfg.SSOSessionName
+				cfg.SSOSession = &ssoSession
```

Comment:
**Question**: So this looks like we set a single `SSoSession` for the entire shared config. How does this work if we have an assume role chain? (this is mostly me trying to understand the existing Go v1 shared config loading).

e.g. with `AWS_PROFILE = foo`

`foo` doesn't directly refer to `session1`, what is the resulting `sharedConfig` struct in Go?

```ini
[profile foo]
role_arn = role1
source_profile = bar

[profile bar]
role_arn = role2
source_profile = baz


[profile baz]
role_arn = role3
sso_account_id = 1234
sso_region = us-east-1
sso_session = session1

[sso-session session1]
...

```

Reply:
Then during step of resolving user credentials from [shared config](https://github.com/aws/aws-sdk-go/blob/d4aaf18d8d7042787830094d94f4018afe81f165/aws/session/credentials.go#LL104C20-L104C20), it will recursively detect source profile and resolve sso credential provider if present in `sharedCfg`

Reply:
Great thanks for clarifying this, that is helpful.